skimage/morphology/_skeletonize.py

```diff
@@ -159,13 +160,25 @@ def skeletonize_2d(image):
            [0, 0, 0, 0, 0, 0, 0, 0, 0]], dtype=uint8)
 
     """
-
     if image.ndim != 2:
         raise ValueError("Zhang's skeletonize method requires a 2D array")
 
+    image = image.astype(bool, order="C")
     return _fast_skeletonize(image)
 
 
+def skeletonize_2d(image):
+    return _skeletonize_2d(image)
+
+
+skeletonize_2d.__doc__ = _skeletonize_2d.__doc__
+skeletonize_2d = deprecate_func(
+    deprecated_version="0.22",
+    removed_version="0.24",
+    hint="Use `skimage.morphology.skeletonize` instead."
+)(skeletonize_2d)
```

Comment:
Why this instead of using `deprecate_func` as a decorator...?

Reply:
`deprecate_func` modifies the docstring and I didn't copy over the docstring to the placeholder `skeletonize_2d` so I'm doing that at runtime before wrapping. I'm not sure why I did it that way but maybe to keep the diff smaller and preserver history for the original docstring.. I can change it if you prefer the other approach.